### PR TITLE
feat: enable sc command for cdk-import

### PR DESCRIPTION
**Pull request by dponzo**
Add entry-point and cli commands for active service catalog import path.

This change adds a subcommand for sc as well as a default path for cfn to maintain consistency.

The sc subcommand supports pulling in products based on a default strategy or by pulling in a product directly by IDs

Co-authored-by: Aidan Crank <arcrank@gmail.com>

### CODE REVIEW

**Comment by dponzo** [comment]
adding some minor tweaks with @arcrank via PR, will request review when ready

**Comment by arcrank** [comment]
Open to any input on how to share the facade of the tool (cx facing outputs). 
For now, keeping it kind of separate with different specific help messages for `cfn` vs `sc`. 

We updated the README docs for service catalog.  Once logic is live and usable, we'll get some rapid feedback on what types of examples are the most useful and we will add that to the README in the Service Catalog section.

Let us know if you have any questions or issues.

File: README.md [hunk 1]
```diff
@@ -140,6 +161,40 @@ Modules are also supported:
 cdk-import AWSQS::CheckPoint::CloudGuardQS::MODULE
 ```
 
+## AWS Service Catalog Usage
+
+The cdk-import tool generates a user friendly version of an provisioned product
```
Comment (dponzo) [inline README.md:166]:
```suggestion
The cdk-import tool generates a user friendly version of a provisioned product
```

**Comment by arcrank** [comment]
Should link : https://github.com/cdklabs/cdk-import/issues/203

`minimist-subcommand` is MIT License as well that we can use.

File: README.md [hunk 2]
```diff
@@ -11,9 +11,30 @@ npm install -g cdk-import
 
 ## Usage
 
+There are currently two sources that resources can be generated from. The subcommand 
+`cfn` is used to import from CloudFormation Registry, 
+`sc` is used to import AWS Service Catalog products.
+There are shared general options for output directories and target language.
+The `cfn` subcommand is optional as it is default import path.
```
Comment (corymhall) [inline README.md:18]: This library is still experimental so I think I am fine with a breaking change here to require a subcommand.

File: README.md [hunk 3]
```diff
@@ -140,6 +161,40 @@ Modules are also supported:
 cdk-import AWSQS::CheckPoint::CloudGuardQS::MODULE
 ```
 
+## AWS Service Catalog Usage
+
+The cdk-import tool generates a user friendly version of a provisioned product
+that can be used like a normal cdk construct within a cdk app.
+You can currently either specify a specific product version or generate all available products
+The tool will call APIs and attempt to resolve default artifact and launch path for a product,
```
Comment (corymhall) [inline README.md:169]: Since SC permissions work differently is it worth calling out what kind of access you need?

File: src/cli.ts [hunk 4]
```diff
@@ -55,38 +91,127 @@ function showHelp() {
...
+      const workdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk-import'));
+      const ppArgs = [args['product-id'], args['provisioning-artifact-id'], args['path-id']].filter(param => param != undefined);
+
+      if (args.help || (ppArgs.length > 0 && ppArgs.length != 3)) {
+        showSCHelp();
+        process.exit(1);
+      }
+
+      if (!args['csharp-namespace'] && args.language == 'csharp') {
+        throw new Error('C# namespace name (`--csharp-namespace`) must be specified (e.g. "AWS::Foo::Bar")');
```
Comment (corymhall) [inline src/cli.ts:179]: is `AWS::Foo::Bar` an actual namespace name?

**Review by corymhall — COMMENTED** [review]
Just a couple of comments.

File: README.md (hunk 3 above)
Comment (dponzo) [reply README.md:169]: sure that is a good idea

File: README.md (hunk 2 above)
Comment (dponzo) [reply README.md:18]: cool, sounds good, wasn't sure if we could do that

File: src/cli.ts (hunk 4 above)
Comment (dponzo) [reply src/cli.ts:179]: thats more to show the format, I was referencing the comment from the cfn registry resource typename

Comment (dponzo) [reply src/cli.ts:179]: okay i see, will fix

Comment (arcrank) [reply src/cli.ts:179]: whoops great catch!  Pulled this from the cfn part, but yeah, we're not deriving from  a type here so the namespace should be a valid one.

Comment (arcrank) [reply src/cli.ts:179]: I don't even think an example is necessary.

Comment (dponzo) [reply src/cli.ts]: ended up removing this and defaulting to the product name, similar to the cfn case

File: README.md (hunk 3 above)
Comment (dponzo) [reply README.md:171]: added a line about it

File: src/cli.ts (hunk 4 above)
Comment (dponzo) [reply src/cli.ts]: thus making csharp namespace optional for both subcommands

**Review by corymhall — APPROVED** [review]

**Pull request merged**